src/main/kotlin/org/ageseries/libage/sim/thermal/Thermal.kt

```diff
@@ -0,0 +1,306 @@
+package org.ageseries.libage.sim.thermal
+
+import org.ageseries.libage.data.mutableMultiMapOf
+import kotlin.math.pow
+import kotlin.math.sqrt
+
+/**
+ * A temperature scale.
+ *
+ * In general, don't construct these unless you really are making a new temperature scale; the associated
+ * constant members should suffice for most use cases.
+ *
+ * All scales are defined relative to Kelvin, which is the package-wide base unit.
+ */
+class Scale(
+    val factor: Double,
+    val base: Double,
+    /** The unit suffix string accepted for use with this temperature scale, to be displayed after the value. */
+    val displayUnit: String = "",
+) {
+    /** Given a temperature [t] in Kelvin, return its value in this unit. */
+    fun map(t: Double): Double = factor * t + base
+    /** Given a temperature [t] in this unit, return its value in Kelvin. */
+    fun unmap(t: Double): Double = (t - base) / factor
+
+    /**
+     * Give a human-readable string of a temperature in this scale.
+     */
+    fun display(t: Double): String = "${t}${displayUnit}"
```

Comment:
Note: You'll need to actually extend this in Eln2 with I18N network serialization, so I'd make this an open class or plan to extend it via an interface or extension function.

Reply:
I do wonder whether or not the _symbols_ are an international standard--not that it hurts, but I think Kelvin is "K" everywhere, even in reports written in other languages :)

I don't know the internationalization system well--but if it's going to be network-serializeable, do you want to just make an `enum class` out of this, throwing extensibility to the wind?

Reply:
Extensibility is important. I just wanted to mention that supporting language features here is not a great idea and that it should be extensible.